Source/HelixToolkit.SharpDX.Shared/Model/Scene/CrossSectionMeshNode.cs

```diff
@@ -238,6 +240,86 @@ protected override RenderCore OnCreateRenderCore()
             {
                 return new CrossSectionMeshRenderCore();
             }
+
+            protected override bool OnHitTest(RenderContext context, Matrix totalModelMatrix, ref Ray rayWS, ref List<HitTestResult> hits)
+            {
+                int hitsBeforeCheck = hits?.Count ?? 0;
+                var meshGeometry3d = Geometry as MeshGeometry3D;
+                if (meshGeometry3d == null)
+                    return false;
+                meshGeometry3d.ReturnMultipleHitsOnHitTest = true;
```

Comment:
Set it back to false after hit test, since MeshGeometry3D can be shared across multiple models.

Reply:
Hmm, in that case there is a risk that one call sets it and then the next one unsets the flag before the first call sets it back. Or are all of these calls guaranteed to be on the same thread?

Reply:
Yes, hit test is in UI thread.